File: CopyCopy.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int t;
    cin >> t;
    vector<int> ans;
    while (t--) {
        int n;
        cin >> n;
        set<int> arr;
        for (int i=0; i<n; i++) {
            int temp;
            cin >> temp;
            arr.insert(temp);
        }
        ans.push_back(arr.size());
    }
    for (auto &x:ans)
        cout << x << endl;

    return 0;
}

File: Python/CopyCopy.py
def main():
    t = int(input())
    ans = []
    while t:
        t -= 1
        n = int(input())
        arr = set()
        for i in range(n):
            temp = int(input())
            arr.add(temp)
        ans.append(len(arr))
    for i in ans:
        print(i)


if __name__ == '__main__':
    main()
